File: XPlaneParser/src/main/java/ms/xplaneparser/entity/Taxiway.java
package ms.xplaneparser.entity;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;
@Getter @Setter
public class Taxiway implements Comparable<Taxiway>  {
	private static final String newLine = System.getProperty("line.separator");
	private String name;
	
	private List<LatLong> nodes=new ArrayList<>();
	private boolean centerLine =false;
	private boolean runwayHold=false;
	
	public void addNode(double latitude,double longitude) {
		nodes.add(new LatLong(latitude,longitude));
	}
	public LatLong getLastNode() {
		if (nodes.isEmpty()) return null;
		return nodes.get(nodes.size()-1);
	}
	public void closeLinearLoop() {
		nodes.add(nodes.get(0));
	}
	@Override
	public String toString() {
		
		StringBuilder sb=new StringBuilder( "--Taxiway ID:  " + name);
		String nodesDescriptor = nodes.stream()
			.map(LatLong::toString)
			.collect(Collectors.joining(newLine));
		sb.append(nodesDescriptor);
		return sb.toString();

	}

	@Override
	public int compareTo(Taxiway t) {
		return name.compareTo(t.getName());
	}
	
	

}

File: XPlaneParser/src/main/java/ms/xplaneparser/entity/Taxiway.kt
package ms.xplaneparser.entity

import java.util.*
import java.util.stream.Collectors

class Taxiway : Comparable<Taxiway> {
    var name: String? = null
    var nodes: MutableList<LatLong> = ArrayList()
    var isCenterLine = false
    var isRunwayHold = false
    fun addNode(latitude: Double, longitude: Double) {
        nodes.add(LatLong(latitude, longitude))
    }

    val lastNode: LatLong?
        get() = if (nodes.isEmpty()) null else nodes[nodes.size - 1]

    fun closeLinearLoop() {
        nodes.add(nodes[0])
    }

    override fun toString(): String {
        val sb = StringBuilder("--Taxiway ID:  $name")
        val nodesDescriptor = nodes.stream()
                .map { obj: LatLong -> obj.toString() }
                .collect(Collectors.joining(newLine))
        sb.append(nodesDescriptor)
        return sb.toString()
    }

    override fun compareTo(t: Taxiway): Int {
        return name!!.compareTo(t.name!!)
    }



    companion object {
        private val newLine = System.getProperty("line.separator")
    }
}